pkg/manager/impl/execution_manager.go

```diff
@@ -511,6 +511,14 @@ func (m *ExecutionManager) getExecutionConfig(ctx context.Context, request *admi
 		if launchPlan.Spec.WorkflowId != nil {
 			workflowName = launchPlan.Spec.WorkflowId.Name
 		}
+		// Backward compatibility changes to get security context from auth role.
```

Comment:
this is getting so hairy :( can you add a more detailed comment why this is necessary?

Reply:
Hi @katrogan added comments for these